phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -883,6 +903,14 @@ public void readFields(DataInput input) throws IOException {
         dataEmptyKeyValueCF = Bytes.readByteArray(input);
         emptyKeyValueCFPtr = new ImmutableBytesPtr(Bytes.readByteArray(input));
         
+        int numIndexedExpressions = WritableUtils.readVInt(input);
```

Comment:
The IndexMaintainer is sent from the client to the server - it's like a compiled index definition. You'll just need to make sure that when you deserialize the data, you can detect that it's from an old client and you can read it correctly to establish the correct state. Remember, you'll know in this case that there are no functional indexes.